### PR TITLE
Added the snowflake datasource

### LINKED ISSUE #46: Add Snowflake Connector
Didn't see Snowflake as one of the cloud data warehouse sources.

### PR DESCRIPTION
**Pull request by joezein-oss**
## PR Goal:
Added the Snowflake datasource connector and performed manual testing.
This PR Closes #46

## Implementation and Testing Details:
- Integrated the Snowflake datasource into the database and backend/frontend code, using the sqlalchemy-snowflake library to maintain consistency with other datasources.
- Tested creating a Snowflake datasource.
- Tested editing the Snowflake datasource.
- Tested pinging the datasource from the frontend (verified both failure and success cases).
- Verified the Snowflake schema functionality.
- Tested creating SQL cells using the Snowflake datasource.
- Tested running queries with the Snowflake datasource.
- Tested stopping query execution.

**Note: I didn't see any PR templates, so I provided details on what I added and tested. If you need additional information or screenshots, feel free to let me know, and I'll be happy to provide them.**

### CODE REVIEW

**Review by defihook — COMMENTED** [review]
looks awesome

**Review by lucasfcosta — COMMENTED** [review]
I need to do some more E2E testing here - but overall it looks good to me. Just a few minor comments in the meantime while I get a snowflake instance to test it.

**Comment by lucasfcosta** [comment]
I went ahead and pushed some updates here so it works and also E2E tested it.

It's good-to-go on my end, but I want @vieiralucas to double check it.

File: apps/web/src/pages/workspaces/[workspaceId]/data-sources/new/snowflake.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  },
+  {
+    name: 'Add Snowflake data source',
+    icon: PlusCircleIcon,
+    href: '#',
+    current: true,
+  },
+]
+
+export default function NewDataSourcePostgresSQLPage() {
```
Comment (lucasfcosta) [inline apps/web/src/pages/workspaces/[workspaceId]/data-sources/new/snowflake.tsx:29]:
```suggestion
export default function NewDataSourceSnowflakePage() {
```

**Review by lucasfcosta — APPROVED** [review]

**Comment by lucasfcosta** [comment]
@joezein-oss merging this! Amazing work, thank you!

**Pull request merged**